runtime/src/result.rs

```diff
@@ -47,6 +47,18 @@ impl<T, E> From<Result<T, E>> for DiplomatResult<T, E> {
     }
 }
 
+impl<T> From<Option<T>> for DiplomatResult<T, ()> {
```

Comment:
what do we need these for? I'm not sure we want to provide transitive conversions, we should try to keep things explicit.

Reply:
That's not transitive, DiplomatOption _is_ DiplomatResult. It's a type def.

Reply:
ah of course, can you use the alias here?